packages/cardano-services/src/util/DbSyncProvider/DbSyncProvider.ts

```diff
@@ -51,34 +43,37 @@ export const DbSyncProvider = <
     }
 
     public async healthCheck(): Promise<HealthCheckResponse> {
+      const response: HealthCheckResponse = { ok: false };
       try {
-        const { ok, localNode } = await this.cardanoNode.healthCheck();
-        const tip = (await this.db.query<LedgerTipModel>(findLedgerTip)).rows[0];
-        const isHealthy =
-          ok && tip.block_no >= (localNode?.ledgerTip?.blockNo ?? DB_MAX_SAFE_INTEGER) - DB_BLOCKS_BEHIND_TOLERANCE;
+        let tip: LedgerTipModel | undefined;
+        const cardanoNode = await this.cardanoNode.healthCheck();
+        response.localNode = cardanoNode.localNode;
+        try {
+          tip = (await this.db.query<LedgerTipModel>(findLedgerTip)).rows[0];
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+          this.logger.error(error.message);
+        }
```

Comment:
Errors thrown by `this.db.query` are caught and logger here, then the tip in the response is set. I decided to not put the try/catch around `this.cardanoNode.healthCheck` invocation, which is primarily the reason for the outer one, since it then ensures this function doesn't throw, which is the goal here.

Reply:
> then the tip in the response is set

If we throw on this [line](https://github.com/input-output-hk/cardano-js-sdk/pull/644/files#diff-4737374e2ac27a76e30bc8c8858b7890aee4d442cffc3fe03ce9e9ebe77b0247R52) with `tip = (await this.db.query()`, then when the tip is set?

Reply:
There's no need to set the tip in a failing output.
Which this change we get an unhealthy response, from the absence of the tip we can assume the source of unhealthy is the unavailability od the DB

Reply:
That's totally right, but I still can not see the reason for the second try-catch, maybe I'm missing something 👀

Reply:
Yep, this can be simplified. Good catch!

Reply:
I had to rebase on master with a few dependent changes merged in other PRs, so needed to force-push. The suggested change is implemented [here](https://github.com/input-output-hk/cardano-js-sdk/pull/644/files#diff-4737374e2ac27a76e30bc8c8858b7890aee4d442cffc3fe03ce9e9ebe77b0247R45-R71)